x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java

```diff
@@ -736,7 +729,7 @@ public static <T> Set<Class<? extends T>> subclassesOf(Class<T> clazz, Predicate
             // NIO FileSystem API is not used since it trips the SecurityManager
             // https://bugs.openjdk.java.net/browse/JDK-8160798
             // so iterate the jar "by hand"
-            if (path.endsWith(".jar") && path.contains("x-pack-ql")) {
```

Comment:
This prevented the `esql.core` classes from being tested and allowed the contract breaches to slip through.